Mathlib/Topology/LocallyConstant/Algebra.lean

```diff
@@ -291,4 +291,63 @@ theorem coe_algebraMap (r : R) : ⇑(algebraMap R (LocallyConstant X Y) r) = alg
 
 end Algebra
 
+section Comap
+
+variable [TopologicalSpace Y]
+
+/-- `LocallyConstant.comap` is a multiplicative map. -/
+noncomputable
+def comapMul  [MulOneClass Z] (f : X → Y) (hf : Continuous f) :
+    LocallyConstant Y Z →* LocallyConstant X Z where
+  toFun := comap f
+  map_one' := by
+    ext x
+    rw [coe_comap_apply _ _ hf]
+    rfl
+  map_mul' r s := by
+    ext x
+    dsimp only [coe_mul, Pi.mul_apply]
+    rw [coe_comap_apply _ _ hf, coe_comap_apply _ _ hf, coe_comap_apply _ _ hf]
+    rfl
+
+/-- `LocallyConstant.comap` is a linear map. -/
+noncomputable
+def comapLinear {R : Type _} [Semiring R] [AddCommMonoid Z] [Module R Z] (f : X → Y)
+    (hf : Continuous f) : LocallyConstant Y Z →ₗ[R] LocallyConstant X Z where
+  toFun := comap f
+  map_add' := by
+    intro r s
+    ext x
+    dsimp only [coe_add, Pi.add_apply]
+    rw [coe_comap_apply _ _ hf, coe_comap_apply _ _ hf, coe_comap_apply _ _ hf]
+    rfl
+  map_smul' r s := by
+    ext x
+    dsimp only [RingHom.id_apply, coe_add, Pi.add_apply, AddHom.toFun_eq_coe,
+      AddHom.coe_mk, coe_smul, Pi.smul_apply]
+    rw [coe_comap_apply _ _ hf, coe_comap_apply _ _ hf]
+    rfl
+
+lemma comapLinear_injective {R : Type _} [Semiring R] [AddCommMonoid Z] [Module R Z] (f : X → Y)
+    (hf : Continuous f) (hfs : Function.Surjective f) :
+    LinearMap.ker (comapLinear f hf : LocallyConstant Y Z →ₗ[R] LocallyConstant X Z) = ⊥ :=
+  LinearMap.ker_eq_bot_of_injective <| comap_injective _ hf hfs
```

Comment:
Usually when we put `injective` we actually use `Function.injective`, the kernel result should be called `ker_comapLinear`. See e.g [LinearMap.lsmul_injective](https://leanprover-community.github.io/mathlib4_docs/Mathlib/LinearAlgebra/BilinearMap.html#LinearMap.lsmul_injective) vs [LinearMap.ker_lsmul](https://leanprover-community.github.io/mathlib4_docs/Mathlib/LinearAlgebra/BilinearMap.html#LinearMap.ker_lsmul)